control-plane/agents/src/bin/core/tests/rebuild/mod.rs

```diff
@@ -169,14 +177,92 @@ async fn replica_wait_and_full_rebuild() {
         .await
         .unwrap();
 
-    let volume_state = volume.state;
-    assert!(volume_state.status == VolumeStatus::Online);
+    let volume_state = volume.state.clone();
+    assert_eq!(volume_state.status, VolumeStatus::Online);
     let nexus = volume_state.target.unwrap();
     assert_eq!(nexus.children.len(), 3);
     nexus
         .children
         .iter()
-        .for_each(|c| assert!(c.state == ChildState::Online));
+        .for_each(|c| assert_eq!(c.state, ChildState::Online));
+
+    // Check where the replicas are, apart from vol target node.
+    let replicas = api_client.replicas_api().get_replicas().await.unwrap();
+    tracing::info!({?replicas}, "Available replicas - ");
+    let testrep = replicas.iter().find(|r| r.node != vol_target).unwrap();
+    let testrep_node = &testrep.node;
+    tracing::info!(
+        "Stopping node {testrep_node} having replica {}",
+        testrep.uri.clone()
+    );
+
+    // Stop the container and don't restart.
+    cluster
+        .composer()
+        .stop(&testrep_node.to_string())
+        .await
+        .expect("container stop failure");
+
+    // Check the pool on restarted node shouldn't be online.
+    // This must expire the wait timeout.
+    wait_till_pool_online(&cluster, testrep, Duration::from_secs(20)).await;
```

Comment:
Seems like we want the opposite instead, wait_till_pool_offline.

Reply:
added validation on the returned result. we werent processing them before. changed fn name to `is_pool_online ` .